storage/src/commit_log/mod.rs

```diff
@@ -0,0 +1,430 @@
+// Copyright (c) SimpleStaking and Tezedge Contributors
+// SPDX-License-Identifier: MIT
+
+mod compression;
+
+use failure::Fail;
+use serde::{Deserialize, Serialize};
+
+use crate::persistent::codec::{Decoder, Encoder, SchemaError};
+use crate::persistent::schema::{CommitLogDescriptor, CommitLogSchema};
+use crate::persistent::BincodeEncoded;
+use std::{fmt, io};
+
+use std::path::{Path, PathBuf};
+use std::sync::{Arc, RwLock};
+
+use crate::commit_log::compression::{zstd_compress, zstd_decompress};
+use std::collections::HashMap;
+use std::fs::{File, OpenOptions};
+use std::io::{BufReader, BufWriter, Read, Seek, SeekFrom, Write};
+
+pub type CommitLogRef = Arc<RwLock<CommitLog>>;
+
+const DATA_FILE_NAME: &str = "table.data";
+
+pub struct CommitLog {
+    data_file: File,
```

Comment:
@mambisi @jurajselep 
Mambisi, this one, I am not really sure, is it ok to have just one big file?
we dont remove from this file (we would like to also, but this commit_log stuff with fixed Location does not support it yet). I am just afraid, when this file will be too long, lets. say 1300k blocks of mainnet?

Reply:
we can chunk it in smaller files, all that need to be updated is the` struct Location` to include the file id that holds the data

Reply:
@mambisi @tizoc 
what do you recommend? 

I am thinking something like this:
```
pub struct Location(pub DbOffset, pub Offset, pub ByteLimit);
```

DbOffset - could be some u64 and it will point to internal CommitLog's map: `Map<DbOffset, fileName>`,
not to save fileName all the time to rocksdb

Reply:
Map<FileId,PathBuf>, where the FileId is cycle offset (1024.log, 2048.log), (1024.log will contain blocks from 0..1024)

Reply:
I am not sure if commit log stores blocks in order